dev/benchmarks/macrobenchmarks/lib/src/web/bench_draw_rect.dart

```diff
@@ -10,27 +10,63 @@ import 'recorder.dart';
 ///
 /// Measures the performance of the `drawRect` operation.
 class BenchDrawRect extends SceneBuilderRecorder {
-  BenchDrawRect() : super(name: benchmarkName);
+  /// A variant of the benchmark that uses the same [Paint] object for all rectangles.
+  ///
+  /// This variant focuses on the performance of the `drawRect` method itself.
+  BenchDrawRect.staticPaint() : isVariablePaint = false, super(name: benchmarkName);
+
+  /// A variant of the benchmark that creates a unique [Paint] for each rectangle.
+  ///
+  /// Does not cache the [Paint] objects across frames, but generates new
+  /// objects every time. This variant of the benchmark focuses on construction
+  /// and transfer of paint data to the renderer.
+  BenchDrawRect.variablePaint() : isVariablePaint = true, super(name: variablePaintBenchmarkName);
 
   static const String benchmarkName = 'draw_rect';
+  static const String variablePaintBenchmarkName = 'draw_rect_variable_paint';
 
   /// Number of rows in the grid.
   static const int kRows = 25;
 
   /// Number of columns in the grid.
   static const int kColumns = 40;
 
+  /// Whether each cell should gets its own unique [Paint] value.
+  ///
+  /// This is used to benchmark the efficiency of passing a large number of
+  /// paint objects to the rendering system.
+  final bool isVariablePaint;
```

Comment:
nit: maybe rename to something clearer like `useUniquePaint`?

Reply:
Done (renamed to `benchmarkPaint` as discussed offline)